### PR TITLE
MediaController in videojs was not releasing some handlers during dispose

### PR DESCRIPTION
**Pull request by saxena-gaurav**
Rebasing the code against stable branch. Credit to author @giakas for his original PR #1939

### CODE REVIEW

File: src/js/media/html5.js [hunk 1]
```diff
@@ -149,6 +149,12 @@ vjs.Html5.prototype.createEl = function(){
 
     vjs.insertFirst(el, player.el());
   }
+  else {
+    // need to make sure we add the video tag back in the UI, when we are changing tech to html5.
+    if (el.parentNode !== player.el()) {
+    vjs.insertFirst(el, player.el());
```
Comment (mmcc) [inline src/js/media/html5.js]: Looks like the indentation is weird here.

File: src/js/media/media.js [hunk 2]
```diff
@@ -89,6 +81,16 @@ vjs.MediaTechController.prototype.initControlsListeners = function(){
   });
 };
 
+/**
+ * Remove the controls handler.
+ */
+vjs.MediaTechController.prototype.removeControls = function () {
```
Comment (mmcc) [inline src/js/media/media.js]: Not a huge fan of this naming, since it's removing *handlers* not controls.

File: src/js/media/media.js [hunk 3]
```diff
@@ -124,6 +126,14 @@ vjs.MediaTechController.prototype.addControlsListeners = function(){
   this.on('tap', this.onTap);
 };
 
+vjs.MediaTechController.prototype.activateControls = function () {
```
Comment (mmcc) [inline src/js/media/media.js]: Same as above, these methods should indicate that they're handling listeners.

**Comment by mmcc** [comment]
Other than the comments above this looks good. After this gets merged to stable we should create a new PR against master and just copy things over. That's going to be waaay easier than trying to do this via a rebase or something.

**Comment by saxena-gaurav** [comment]
@mmcc : Thanks for the review. Addressed the feedback. Once this is merged to stable, I will update the PR #2053 (PR against the master branch).

File: src/js/media/html5.js [hunk 4]
```diff
@@ -149,6 +149,12 @@ vjs.Html5.prototype.createEl = function(){
 
     vjs.insertFirst(el, player.el());
   }
+  else {
```
Comment (heff) [inline src/js/media/html5.js]: `else` should be on the same line as the previous curly brace here. `} else {`

File: src/js/player.js [hunk 5]
```diff
@@ -1430,6 +1430,8 @@ vjs.Player.prototype.error = function(err){
     return this.error_;
   }
 
+  if (this.tech) { vjs.MediaTechController.prototype.stopTracking.call(this.tech); }
```
Comment (heff) [inline src/js/player.js]: I think we could just call `this.tech.stopTracking()` here.

**Comment by saxena-gaurav** [comment]
@heff : Thanks for the code changes.

**Comment by saxena-gaurav** [comment]
Closing in favor of #2095

**Pull request closed without merging**